deepspeed/runtime/engine.py

```diff
@@ -808,6 +808,10 @@ def communication_data_type(self):
 
         return torch.float32
 
+    @communication_data_type.setter
```

Comment:
The communication data type should be set via ds_config rather than hardcoding for SP>1. Below is the guide. 

https://github.com/microsoft/DeepSpeed/blob/e238351101bcd23a9a0a32dba0b071685d670d6c/deepspeed/runtime/constants.py#L213-L224